src/profile/v100/profile.rs

```diff
@@ -390,6 +415,24 @@ mod tests {
         assert_eq!(set.len(), n);
     }
 
+    #[test]
+    fn to_json_bytes_new_from_json_bytes() {
+        let sut = SUT::sample();
+
+        let encoded = sut.to_json_bytes().unwrap();
+        let profile_result = SUT::new_from_json_bytes(encoded).unwrap();
+        assert_eq!(profile_result, sut);
+
+        let malformed_profile_snapshot = BagOfBytes::from("{}".as_bytes());
```

Comment:
can you split the happy path and unhappy path unit tests in two independent tests?